test/cmd/images-old-policy.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${OS_ROOT}/hack/lib/init.sh"
+os::log::stacktrace::install
+trap os::test::junit::reconcile_output EXIT
+
+# Cleanup cluster resources created by this test
+(
+  set +e
+  oc login -u system:admin
+  oadm policy reconcile-cluster-roles --additive-only=false --confirm
+  exit 0
+) &>/dev/null
+
+
+os::test::junit::declare_suite_start "cmd/images-old-policy"
+
+oc project -q
+project=$(oc project -q)
```

Comment:
oc project -q should never change project.  Ever.

On Wed, Aug 3, 2016 at 1:27 PM, David Eads <notifications@github.com> wrote:

> In test/cmd/images-old-policy.sh
> <https://github.com/openshift/origin/pull/8497#discussion_r73373129>:
>
> > +os::log::stacktrace::install
> > +trap os::test::junit::reconcile_output EXIT
> > +
> > +# Cleanup cluster resources created by this test
> > +(
> > +  set +e
> > +  oc login -u system:admin
> > +  oadm policy reconcile-cluster-roles --additive-only=false --confirm
> > +  exit 0
> > +) &>/dev/null
> > +
> > +
> > +os::test::junit::declare_suite_start "cmd/images-old-policy"
> > +
> > +oc project -q
> > +project=$(oc project -q)
>
> we enter this test in the correct project, but the oc login -u
> system:admin on line 15 lists available projects, and switches to "default"
> if the current project is not in the list. That races with the project
> authz cache seeing the new project
>
> Huh? That should run a sub-shell, which runs those commands, outputs both
> streams to dev/null and blocks until complete. The & is tied to &>, so
> its a redirect, not a spawn.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/8497/files/6e2823385e9eb13d67b3be6443b644dacb9f08b0#r73373129>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p3KBCtdxjlUXgfGFe6c1xnS8csmRks5qcM-WgaJpZM4IGbe8>
> .
>